server/pbench/bin/index-pbench

```diff
@@ -321,6 +368,115 @@ class ResultData(object):
             datafiles.append(datafile)
         return datafiles
 
+    @staticmethod
+    def expand_template(templ, d):
+        return templ
+
+    @staticmethod
+    def make_datum_wrapper(templ):
+        # make a shallow copy
+        d = templ.copy()
+        del d['timeseries']
+        # construct the uid from the template and the values in d
+        # and replace the template with the result
+        d['uid'] = ResultData.expand_template(d['uid'], d)
```

Comment:
Is this going to be an MD5 sum of the timeseries?  Not sure what this logic is about.

Reply:
The `uid` element is going to be the expanded template, replacing `%foo%` in the template with the value of `foo` in the dict `d`. The MD5 sum over the whole doc is going to be the source id of the document. See commit  [ 58d8abf ](https://github.com/distributed-system-analysis/pbench/pull/959/commits/58d8abf45fe6bdaffb8f3dd261861866ccd2f3c5) for the source id mod and commit [e8f89dc](https://github.com/distributed-system-analysis/pbench/pull/959/commits/e8f89dc84e170ad6652f6980ae14df55d64a4666) for the template expansion mod.